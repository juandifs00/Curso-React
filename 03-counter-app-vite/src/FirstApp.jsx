import PropTypes from 'prop-types';

const getResult = (a, b) => {
  return (a*b)/(a+b);
}

export const FirstApp = ( {title, subTitle, name} ) => {

  //console.log(props);

  return (
    <>
      {/*<h1>Hello, I'm Juandifs</h1>*/}
      <h1>{ title }</h1>
      {/*<h1>Resultado: {getResult(15,45)}</h1>*/}
      {/*<code> { JSON.stringify(newMessaje) } </code>*/}
      <p>Soy algo que salío aquí</p>
      <p>{subTitle}</p>
      <p>{name}</p>
    </>
  )
}
  

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string
}

FirstApp.defaultProps = {
  name: 'Juan Diego RG',
  subTitle: 'No hay un subtítulo',
  //title: 'No hay un título',
}