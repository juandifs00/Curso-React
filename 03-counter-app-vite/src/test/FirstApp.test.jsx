import { render } from "@testing-library/react"
import { FirstApp } from "../FirstApp"

describe('Pruebas en FirstApp', () => { 
    
    test('Debe hacer match en el snapshot', () => { 
        
        const title = 'Hola, soy Juandi';
        const { container } = render( <FirstApp title={ title } /> )

        expect ( container ).toMatchSnapshot();

     })

     test('Debe de mostrar el titulo en un h1', () => { 
        
        const title = 'Hola, soy Juandi';
        const { container, getByText } = render( <FirstApp title={ title } /> )

        expect( getByText( title ) ).toBeTruthy();

        const h1 = container.querySelector( 'h1' );
        
        expect(h1.innerHTML).toContain( title );

      })

 })