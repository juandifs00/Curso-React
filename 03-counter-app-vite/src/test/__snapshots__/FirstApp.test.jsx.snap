// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp Debe hacer match en el snapshot 1`] = `
<div>
  <h1>
    Hola, soy Juandi
  </h1>
  <p>
    Soy algo que salío aquí
  </p>
  <p>
    No hay un subtítulo
  </p>
  <p>
    Juan Diego RG
  </p>
</div>
`;
